tests/integration/test_create_token.py

```diff
@@ -0,0 +1,138 @@
+        },
+        "payload": {
+            "version": AuthPayloadVersion.V2.value,
+            "survey_metadata": {
+                "data": {
+                    "case_ref": "1000000000000001",
+                    "qid": str(uuid.uuid4()),
+                },
+                "receipting_keys": ["qid"],
+            },
+            "collection_exercise_sid": "789",
+            "response_id": "1234567890123456",
+            "language_code": "en",
+            "roles": [],
+            "account_service_url": ACCOUNT_SERVICE_URL,
+        },
+    },
+]
+
+"""
+The purpose of this test class is to test the creation of a token (from create_token.py) to ensure metadata in
+a decrypted token is nested/found in the correct level. create_token_v2 is used as v1 methods are intended to
+be dropped and the other tests only use methods that do not have a v2 implementation.
+"""
+
+
+class TestCreateToken(IntegrationTestCase, AppContextTestCase):
+    def test_payload_from_token(self):
+        for value in test_parameters:
+            with self.subTest():
+                additional_payload = value.get("additional_payload")
+                token = self.token_generator.create_token_v2(
+                    schema_name=value.get("schema"),
+                    theme=value.get("theme"),
+                    **additional_payload,
+                )
+                with self.test_app.app_context():
+                    decrypted_token = decrypt_token(token)
+                    self.assertEqual(
+                        decrypted_token, value.get("payload") | decrypted_token
+                    )
+                    assert (
+                        additional_payload.items()
+                        <= decrypted_token.get("survey_metadata").get("data").items()
+                    )
+
+    def test_uuid_consistent_after_decryption(self):
+        token = self.token_generator.create_token_v2(
+            "test_checkbox.json", theme="social", value="Dummy Text"
+        )
+        with self.test_app.app_context():
+            decrypted_token = decrypt_token(token)
+            assert decrypted_token.get("survey_metadata") == {
+                "data": {
+                    "case_ref": "1000000000000001",
+                    "qid": PAYLOAD_V2_SOCIAL.get("survey_metadata")
+                    .get("data")
+                    .get("qid"),
+                    "value": "Dummy Text",
+                },
+                "receipting_keys": ["qid"],
+            }
+
+    def test_supplementary_data_included_in_token(self):
+        token = self.token_generator.create_supplementary_data_token(
+            "test_checkbox.json", flag_1=True
+        )
+        with self.test_app.app_context():
+            decrypted_token = decrypt_token(token)
+            self.assertEqual(
+                decrypted_token, PAYLOAD_V2_SUPPLEMENTARY_DATA | decrypted_token
+            )
+            assert ({"flag_1": True}).items() <= decrypted_token.get(
```

Comment:
Why are we checking for flag 1 in this test? Should that be separate to supplementary data?

Reply:
Just to test any additional payload is added in at the correct level and I added it as a parameter where I called the `token_generator.create_supplementary_data_token`, happy to change to a different piece of metadata

Reply:
I guess my point is that it is just confusing what is being tested here as the naming of the test suggests it is only checking that the supplementary data metadata is set

Reply:
Maybe to avoid a convoluted name, it maybe be easier to split the test into two, one being `test_supplementary_dataset_included_in_token` and the other being `test_structure_of_supplementary_dataset_token`